sdk/maps/Azure.Maps.Search/src/MapsSearchClient.cs

```diff
@@ -121,768 +118,71 @@ public MapsSearchClient(AzureSasCredential credential, MapsSearchClientOptions o
-        /// Please refer to <see href="https://docs.microsoft.com/azure/azure-maps/supported-languages">Supported Languages</see> for details.
+        /// <param name="top"> Maximum number of responses that will be returned. Default: 5, minimum: 1 and maximum: 20. </param>
+        /// <param name="query"> A string that contains information about a location, such as an address or landmark name. </param>
+        /// <param name="addressLine">
+        /// The official street line of an address relative to the area, as specified by the locality, or postalCode, properties. Typical use of this element would be to provide a street address or any official address.
+        ///
+        /// **If query is given, should not use this parameter.**
         /// </param>
-        /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual async Task<Response<PointOfInterestCategoryTreeResult>> GetPointOfInterestCategoryTreeAsync(SearchLanguage? language = null, CancellationToken cancellationToken = default)
-        {
-            using var scope = _clientDiagnostics.CreateScope("MapsSearchClient.GetPointOfInterestCategoryTree");
-            scope.Start();
-            try
-            {
-                return await RestClient.GetPointOfInterestCategoryTreeAsync(JsonFormat.Json, language.ToString(), cancellationToken).ConfigureAwait(false);
-            }
-            catch (Exception e)
-            {
-                scope.Failed(e);
-                throw;
-            }
-        }
-
-        /// <summary>
-        /// Get POI Category Tree API. POI Category API provides a full list of supported Points of Interest (POI) categories and subcategories together with their translations and synonyms. The returned content can be used to provide more meaningful results through other Search Service APIs, like <see href="https://docs.microsoft.com/rest/api/maps/search/getsearchpoi">Get Search POI</see>.
-        /// </summary>
-        /// <param name="language">
-        /// Language in which search results should be returned. Should be one of supported IETF language tags, except NGT and NGT-Latn. Language tag is case insensitive. When data in specified language is not available for a specific field, default language is used (English).
-        /// Please refer to <see href="https://docs.microsoft.com/azure/azure-maps/supported-languages">Supported Languages</see> for details.
+        /// <param name="countryRegion">
+        /// Signal for the geocoding result to an [ISO 3166-1 Alpha-2 region/country code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) that is specified e.g. FR./
+        ///
+        /// **If query is given, should not use this parameter.**
+        /// </param>
+        /// <param name="bbox">
```

Comment:
We need to change all the `bbox` to `BoundingBox` and use the type `GeoBoundingBox` from `Azure.Core.GeoJson`. Actually we should use all the types in `Azure.Core.GeoJson`

Reply:
Since in generated file SreachRestClient we don't use this type for BoundingBox should we change spec to use that or make converter from Azure.Core.GeoJson.GeoBoundingBox to IEnumerable<double>?
@dubiety

Reply:
Actually, that's we should do (convert it by our own) in our convenience layer in MapsSearchClient.cs
Please see how we did it in our original SDK: https://github.com/Azure/azure-sdk-for-net/blob/5f24bfc2a51a951b61f26968c229a356ab4a7c2d/sdk/maps/Azure.Maps.Search/src/MapsSearchClient.cs#L210
And we can hide unnecessary API like this: https://github.com/Azure/azure-sdk-for-net/blob/5f24bfc2a51a951b61f26968c229a356ab4a7c2d/sdk/maps/Azure.Maps.Search/src/Models/BoundingBox.cs

Reply:
You can refer the [online doc](https://github.com/Azure/autorest.csharp?tab=readme-ov-file#customizing-the-generated-code) to hide, change interface or more customization via annotation using AutoRest.

Reply:
@dubiety done